README.md

```diff
@@ -270,6 +270,86 @@ If there's some error during compilation follow these steps:
+
+- It is assumed that you, the reader, have enough knowledge of bash specifically in order to understand some basic terminology. If not, please look up what you do not know.
+- It is highly recommended to look through the [compile script](build/media-suite_compile.sh) for the exact package you want to modify and cross-reference the functions with the [helper script](build/media-suite_compile.sh) to see what is actually being done.
+- Although it is recommended to use the functions provided in the helper script, there is nothing stopping you from using the actual commands.
+
+--------
+
+- To use a custom patch, create a script within the build folder named `<repository's name>_extra.sh`.
+  - For ffmpeg, the folder name would be ffmpeg-git, but the repository's name would be ffmpeg.
+  - For game-music-emu, the folder name would be game-music-emu-0.6.2 and the repository's name would be game-music-emu
+
+- The commands will run in the cloned/unzipped folder unless otherwise stated. (flac_extra.sh would run commands within flac-git)
+
+To reference the cloned folder itself (`flac-git` `ffmpeg-git`) you can use the variable `${REPO_DIR}`.
+To reference the generated build folder, you would need to use `${REPO_DIR}/build-${bits}` to reference the build-64bit folder for 64 bit builds or build-32bit folder for 32 bit builds.
+
+- For some packages, the build folder may be slightly different (`build-shared-${bits}` for shared ffmpeg or `build-light-${bits}` for light build, etc. Try to compile at least once to see what folders are generated or look through the [compile script](build/media-suite_compile.sh))
+
+If you are building for both 32 and 64-bit, you can gate certain commands using `if [ "$bits"="64bits" ]; then <command>; fi` to only run them when building for 64 bits.\
+Be careful with certain packages due to their unique build process (x265, x264, and some others) so make sure to check the compile script
+
+Example Script: `/build/aom_extra.sh` for `aom-git`
+
+``` bash
+#!/bin/bash
+
+# Commands to run before running cmake
+_pre_cmake(){
+    # Installs libwebp
+    do_pacman_install libwebp
+    # Downloads the patch and then applies the patch
+    do_patch "https://gist.githubusercontent.com/1480c1/9fa9292afedadcea2b3a3e067e96dca2/raw/50a3ed39543d3cf21160f9ad38df45d9843d8dc5/0001-Example-patch-for-learning-purpose.patch"
+}
+
+_post_cmake(){
+    # Change directory to the build folder
+    cd_safe "build-${bits}"
+    # Rerun cmake with custom options. This will override the previous cmake commands.
+    # $LOCALDESTDIR refers to local64 or local32
+    cmake .. -G"Ninja" -DCMAKE_INSTALL_PREFIX="$LOCALDESTDIR" \
+        -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=clang \
+        -DBUILD_SHARED_LIBS=off -DENABLE_TOOLS=off
+}
+
+# Commands to run before building using ninja
+_pre_ninja(){
+    # Change directory to the build folder
+    cd_safe "build-${bits}"
+    # applies a local patch
```

Comment:
I suggest to change to "# Applies to a local file (absolute path)".

And in the previous comment "# Change to the directory of the build folder (relative path)".

It's necessary to clarify: how to use absolute and relative pathes... and what's the root of the relative path (in fact where the file `media-autobuild_suite.bat` lives.

Good job!

Reply:
> how to use absolute and relative pathes... 

We're not teaching people how to use bash here. That's out of scope for the suite.

Reply:
Sorry @wiiaboo !

The idea is to provide _one_ example. Without this info we don't understand where is the root relative path when we're writing these patch scripts.

Please excuse me if I have offended you in any way, or what I have said seems foolish to you.

Reply:
I'm not offended. I'm just telling that we won't be teaching about absolute or relative paths.
The PWD will depend on which points you're at in the build and what package you're building. Most packages will spend most of the time inside the build-${bits}bits directory, but not for FFmpeg, mpv, etc. You can't document that to exhaustion without repeating yourself.

Reply:
Hi @wiiaboo ,

OK. But my assumption (sorry if I'm wrong) is that _all_ patch scripts (for the @1480c1 extension) will have the same syntax. So, in this case _all_ relative pathes point to the same directory, and they not be specific for each project. I'm correct about this assumption?

Reply:
They will be specific to the project.
For flac_extra.sh, they will run in flac-git
For aom_extra.sh, they will run in aom-git